pwnlib/util/cyclic.py

```diff
@@ -344,3 +331,49 @@ def _gen_find(subseq, generator):
         if saved == subseq:
             return pos
     return -1
+
+def _join_sequence(seq, alphabet):
+    if isinstance(alphabet, six.text_type):
+        return ''.join(seq)
+    elif isinstance(alphabet, bytes):
+        return bytes(bytearray(seq))
+    else:
+        return seq
+
+class cyclic_gen(object):
+    def __init__(self, alphabet = None, n = None):
+        if n is None:
+            n = context.cyclic_size
+
+        if alphabet is None:
+            alphabet = context.cyclic_alphabet
+
+        self._generator = de_bruijn(alphabet, n)
+        self._alphabet = alphabet
+        self._total_length = 0
+        self._n = n
+        self._chunks = []
+
+    def get(self, length = None):
+
+        if length != None:
+            self._chunks.append(length)
+            self._total_length += length
+            if len(self._alphabet) ** self._n < self._total_length:
+                log.error("Can't create a pattern length=%i with len(alphabet)==%i and n==%i" \
+                    % (self._total_length, len(self._alphabet), self._n))
+        else:
+            self._chunks.append(float("inf"))
+
+        out = [next(self._generator) for _ in range(length)] if length != None else [next(self._generator)] + list(self._generator)
```

Comment:
You could also use [iters.take](/Gallopsled/pwntools/blob/2116bf3118612a5f53a9a02a8d46ed2f8b1cefcb/pwnlib/util/iters.py#L72) here I guess, as it works also with `n=None`.